### PR TITLE
V2 Basic demo setup for state management

### PR DESCRIPTION
**Pull request by quietbits**
State management setup with a basic demo. I will clean up the demo pieces in the next PR. This is to validate the approach and setup.

Using:
- [Zustand](https://docs.pmnd.rs/zustand/getting-started/introduction) for global state management
- [zustand-querystring](https://github.com/nitedani/zustand-querystring) to create shareable link
  - I went through various options (mostly custom ones), and this one was the best. If something doesn't work with more complex data, we can issue a PR with the fix or fork it.

> [!NOTE]  
> Routing is not working when the page is refreshed because we're using static build. Ops will create a [Node server deployment](https://github.com/stellar/ops/issues/2846) that should fix this issue.

### CODE REVIEW

File: src/store/createStore.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      nestedValue2: number;
+    };
+    update: (value: string) => void;
+    updateNested: (nestedVal: {
+      nestedValue1: string;
+      nestedValue2: number;
+    }) => void;
+    reset: () => void;
+  };
+}
```
Comment (quietbits) [inline src/store/createStore.ts:32]: We can extract sections into the variables if that's easier to manage.

File: src/store/createStore.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        },
+      })),
+      {
+        url: options.url,
+        // Select what to save in query string
+        select() {
+          return {
+            network: true,
+            account: true,
+          };
```
Comment (quietbits) [inline src/store/createStore.ts:78]: This part controls what we want to keep in the URL

**Review by aristidesstaffieri — APPROVED** [review]
yay zustand + immer ❤️

File: src/store/createStore.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        },
+        key: "||",
```
Comment (jeesunikim) [inline src/store/createStore.ts:80]: do we want this vs. its default `"$"`, @quietbits ?

Comment (quietbits) [reply src/store/createStore.ts:80]: @jeesunikim I thought it was a safer option because we could have `$` somewhere (amount in dollars. for example).

**Pull request merged**